rastervision_pytorch_backend/rastervision/pytorch_backend/pytorch_learner_backend_config.py

```diff
@@ -1,94 +1,45 @@
-from typing import List, Optional, Union
-from pydantic import PositiveInt
+from typing import Optional
 
-from rastervision.pipeline.config import (register_config, Field, validator)
+from rastervision.pipeline.config import (register_config, Field)
 from rastervision.core.backend import BackendConfig
+from rastervision.core.rv_pipeline import RVPipeline
 from rastervision.pytorch_learner.learner_config import (
-    SolverConfig, ModelConfig, default_augmentors, augmentors as
-    augmentor_list, PlotOptions, Proportion)
-from rastervision.pytorch_learner.utils import validate_albumentation_transform
+    SolverConfig, ModelConfig, DataConfig, ImageDataConfig)
 
 
 @register_config('pytorch_learner_backend')
 class PyTorchLearnerBackendConfig(BackendConfig):
     model: ModelConfig
     solver: SolverConfig
+    data: DataConfig
```

Comment:
Do you mean retaining all the existing fields in the learner backend config and only adding data as an optional field? That would work, but it would end up creating 2 alternate ways to specify some of the fields in the data config and we will then need some more checks to ensure that they are only specified once, which would add some additional complexity too.

I like the upgrader method because it makes the backward-compatibility-related complexity explicit and keeps it separate from the config itself.

Reply:
Speaking of backward-compatibility, we only care about compatibility with previous releases and not unreleased intermediate versions, right? If so, I think we can safely move `channel_display_groups`, `img_format`, and `label_format` from the SS pipeline config to the data config. But that is probably better done in a separate PR.

Reply:
It would also be nice to have a backward compatibility integration test.

Reply:
Ok, I think I understand things better now, and the current approach you have sounds better. Backward compatibility should be maintained for any model bundle created with any commit after (and including) the 0.12 release. This is because we may have model bundles that were created with non-release commits that need to work with the latest RV in some deployed application. I agree about the backward compatibility integration test, for which we have an issue: https://github.com/azavea/raster-vision/issues/566